### ISSUE TITLE
Harbinger of Doom ability still buged

### ISSUE THREAD

**Issue by k0ut4l1** [WarEmu/WarBugs#15181]
HoD is not working unless you enable/disable it and bring it back to your desire mode. In the video i recorded the healing mode of it but it has the same issue if you have it on the dps mode. It bugs in every zone change. Same as the closed issue #13676 as far as i can tell which was fixed with patch from 03.08.2019.
 
https://www.youtube.com/watch?v=8VF0i8Zk2YM&feature=youtu.be

**Comment by RoR-Zomega** [comment]
Some reproduction steps. You need to have a zealot with some gear that has willpower and int on it (doesn't need to be on the same item) and the same for magic crit and heal crit. I haven't tested magic power as I don't have any equipment with that stat.

When changing zone with the ability turned on: willpower, healing power and heal crit from items is correctly removed from your stats but it is not added to intel / magic power / magic crit. The paper doll says the stats have been added but they are missing from .getstats

When changing zone with the ability turned off: intel and magic crit (magic power untested) from items is correctly removed from your stats but it is not added to willpower / heal crit. The paper doll says the stats have been added but they are missing from .getstats

Basically when transitioning only half of the process is being followed - the bonuses are removed from the correct stats but they are not added to their opposite version. Toggling the ability on/off after each zone change is the only way to ensure your stats are correct.

**Comment by RoR-Zomega** [comment]
New bug to add to the bug: magic crit set bonuses often require you to now unequip and re-equip enough set pieces to remove and add the crit bonus for it to get converted by HoD properly. Example as if I had HoD turned off (healer mode) with an armour set that had a magic crit set bonus.

1) Change zone.
2) Turn HoD on and off to fix conversion issues (post above mine)
3) Do .getstats and see that the magic crit set bonus doesn't show up under magic crit or heal crit
4) Unequip enough items to remove the set bonus
5) Requip items
6) Do .getstats. The magic crit bonus is now being converted to heal crit.

I'm currently experiencing this regularly with 6 piece DPS/alt Sovereign. The worst part is when you unequip and requip you lose and regain the +2 mastery points which with my spec means I lose Wind of Insanity when I unequip the pieces but I don't get it back when I requip them unless I visit a trainer and rebuy Wind of Insanity so this bug is also breaking mastery specs with certain sets.

**Comment by Zomega-WAR** [comment]
Load character. I have a 31 intelligence dagger and a 31 willpower charm equipped and nothing else. Game is correctly granting +62 to willpower and +0 to intelligence. Getstats shows the same as the character sheet.

I then unequip the dagger and the charm. Getstats removes the +62 bonus to willpower but there is still a phantom +31 applied on the character sheet.

![image](https://user-images.githubusercontent.com/60266177/121764407-87653d00-cb3b-11eb-898a-ec50f7c648d1.png)

Toggling Harbinger fixes this.

**Comment by Zomega-WAR** [comment]
Test 2

1) remove all items and toggle harbinger on/off to clear the character sheet
2) turn on harbinger
3) equip the 31 intel dagger and the 31 willpower charm
4) check character sheet. It is displaying +31 to intel and +31 to willpower instead of +62 to intel and +0 to willpower

![image](https://user-images.githubusercontent.com/60266177/121764467-0e1a1a00-cb3c-11eb-9b2e-92ad418a69c2.png)

**Comment by Zomega-WAR** [comment]
Test 3

1) Harbinger on: unequipping items with Willpower won't amend intelligence on the character sheet until you toggle Harbinger
2) Harbinger off: unequipping items with Intelligence won't amend willpower on the character sheet until you toggle Harbinger

Basically, the updating of the character sheet with the stat changes from Harbinger is mostly broken until you toggle modes.

**Comment by Zomega-WAR** [comment]
Crit chance is still being lost when changing zones until you toggle on/off. Confirmed with .getstats so it's a server issue. Example:

1) Gear with 7% heal crit and 9% magic crit on the stats
2) Change zone with Harbinger off. Getstats shows just 7% heal crit instead of 16%.
3) Toggling Harbinger updates the crit rate properly on getstats.

**Comment by Zomega-WAR** [comment]
Today's new test.

Renown: Focused Power IV (+14% magical crit chance)

Event Slot: 1% healing crit
Helm: 2% healing crit (Sentinel)
Shoulders: 2% healing crit (Sovereign Fatemaker set)
Back: 2% healing crit (Sovereign Fatemaker set)
Body: no crit
Gloves: no crit
Belt: no crit
Boots: 2% healing crit (Triumphant)
Main hand: 3% healing crit (Fortress Slicer)
Offhand: 2% magic crit (Bloodlord Focus)
Set bonus: 6% magic crit (Sovereign Fatemaker set)

**With Harbinger off this should be 20% healing crit and 14% magic crit (Harbinger doesn't affect renown crit).**

Character sheet is correct:
![image](https://user-images.githubusercontent.com/60266177/158247253-52885bf4-3a6a-43f5-9e64-2eeab48421a3.png)

.getstats is wrong (missing 8% heal crit):
![image](https://user-images.githubusercontent.com/60266177/158247421-c723bb97-f0d4-4634-a8c1-fd749dff0afa.png)

**With Harbinger on, this should be 0% healing crit and 34% magic crit**

Character sheet is incorrect (failed to convert 2% heal crit):
![image](https://user-images.githubusercontent.com/60266177/158247604-51641a3d-3317-4d4c-813b-beae271859f3.png)

.getstats is wrong (missing 8% magic crit, no heal crit was listed):
![image](https://user-images.githubusercontent.com/60266177/158247715-7219f49e-8eb4-4f7f-aaf8-14263d75657d.png)

It feels like with every patch Harbinger gets worse and worse.
